### PR TITLE
TBOX-286: Add attribute configuration function

### PR DESCRIPTION
**Pull request by hduckduck**
# ↪️ Pull Request

## ✔️ PR Todo

- [ ] Add documentation
- [ ] Add examples
- [ ] Ensure you are following the practices in the [contributor guide](https://docs.google.com/document/d/1c_hhpDMhO0zN793Z_HFOz42Q4g7l1nPRVQ5pXOUsMJU/edit) and [coding_standards](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards)
  * [Use consistent and descriptive name conventions](https://docs.google.com/document/d/1c_hhpDMhO0zN793Z_HFOz42Q4g7l1nPRVQ5pXOUsMJU/edit#heading=h.477s36w4rds9) 
  * [Uses google-style docstrings](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Google-style-docstrings)
  * [Uses versioned API/client methods only](https://docs.google.com/document/d/1c_hhpDMhO0zN793Z_HFOz42Q4g7l1nPRVQ5pXOUsMJU/edit#heading=h.yv6df99fyrq2)
  * [Avoid global variables](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Avoid-global-variables)
  * [Avoid giant try/catch](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Avoid-giant-try-/-except)
  * [Don't use mutable default variables](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Don%E2%80%99t-use-mutable-default-arguments)
  * [Use a `main()` function in your scripts](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Use-a-%E2%80%9Cmain()%E2%80%9D--function-in-your-scripts)
- [ ] Added/updated testing for this change (ensure you are testing any changes/new code!)
- [ ] Update title to link to TBOX issue (i.e. start pull request title with TBOX-xx) 
- [ ] Approval from first code-reviewer
- [ ] Approval from second code-reviewer (only required if a second reviewer has left comments)
- [ ] Pass all checks

### CODE REVIEW

File: tests/project/_common/test_attribute_configuration.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+)
+PROJECT_ID = CONFIG["projects"]["attribute_configuration_editing"]
+
+
+@mock_api()
+def test_update_attribute_configuration_no_failures():
+    client = utils.client.create(**CONFIG["toolbox_test_instance"])
+    project = client.projects.by_resource_id(PROJECT_ID)
+
+    #
```
Comment (wpmartin) [inline tests/project/_common/test_attribute_configuration.py]: typo?

Comment (hduckduck) [reply tests/project/_common/test_attribute_configuration.py]: Yep, corrected it!

File: tamr_toolbox/project/_common/attribute_configuration.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+def get_attribute_configurations(project: Project) -> AttributeConfigurationSpec:
+    """Store the attribute configurations in a list for a given project
+
+    Args:
+        project: Project containing attributes with configurations defined
+
+    Returns:
+        List of AttributeConfigurationSpec
+    """
+
```
Comment (wpmartin) [inline tamr_toolbox/project/_common/attribute_configuration.py]: extra newline not needed

File: tamr_toolbox/project/_common/attribute_configuration.py [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    attribute_configuration_spec = [
+        attribute_configuration.spec() for attribute_configuration in attribute_configuration_all
+    ]
+
+    return attribute_configuration_spec
+
+
+def update_attribute_configuration(
+    project: Project,
+    attributeName: str,
```
Comment (wpmartin) [inline tamr_toolbox/project/_common/attribute_configuration.py]: argument names should be lowercase, e.g. similarity_function

File: tests/project/_common/test_attribute_configuration.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    project = client.projects.by_resource_id(PROJECT_ID)
+
+    with pytest.raises(ValueError):
+        attribute_configuration.update_attribute_configuration(
+            project=project, attributeName="attribute_1", similarityFunction="invalid"
+        )
+
+
+@mock_api()
+def test_update_attribute_configuration_invalid_enabledForMl():
```
Comment (wpmartin) [inline tests/project/_common/test_attribute_configuration.py]: lowercase here too

File: tests/project/_common/test_attribute_configuration.py [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+    project = client.projects.by_resource_id(PROJECT_ID)
+
+    with pytest.raises(ValueError):
+        attribute_configuration.update_attribute_configuration(
+            project=project, attributeName="attribute_1", attributeRole="invalid"
+        )
+
+
+@mock_api()
+def test_update_attribute_configuration_invalid_similarityFunction():
```
Comment (wpmartin) [inline tests/project/_common/test_attribute_configuration.py]: lose the capital letters

File: tests/project/_common/test_attribute_configuration.py [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+    project = client.projects.by_resource_id(PROJECT_ID)
+
+    with pytest.raises(ValueError):
+        attribute_configuration.update_attribute_configuration(
+            project=project, attributeName="attribute_1"
+        )
+
+
+@mock_api()
+def test_update_attribute_configuration_invalid_attributeRole():
```
Comment (wpmartin) [inline tests/project/_common/test_attribute_configuration.py]: lose the cap' R

**Review by wpmartin — APPROVED** [review]
lgtm

**Pull request merged**